student/utils/studentList.js

```diff
@@ -38,15 +38,18 @@ app.controller("myCtrl", function($scope, $filter,$http) {
 		var Email = $scope.addemail;
 		var PhoneNo = $scope.addphone;
 		var Fees = $scope.addfees;
-		if (!BatchId||!Studentname){alert("Please fill all the details");return;} 
+
+		if(!Email){Emial="";}
```

Comment:
its Emial here. That is why it is coming as undefined. make it as email. try adding one new row without email and refresh. to verify if undefined is not coming.